src/jit/morph.cpp

```diff
@@ -11496,7 +11501,16 @@ GenTreePtr Compiler::fgMorphSmpOp(GenTreePtr tree, MorphAddrContext* mac)
                         }
                     }
                 }
-                fgMorphRecognizeBoxNullable(tree);
+
+                __fallthrough;
+
+            case GT_GT:
+
+                if ((oper != GT_GT) || ((tree->gtFlags & GTF_UNSIGNED) != 0))
```

Comment:
Maybe get rid of the fall through and duplicate the call might make this safer and clearer,